client.go

```diff
@@ -832,6 +884,8 @@ func (c *Client) drainBody(body io.ReadCloser) {
 	if err != nil {
 		if c.logger() != nil {
 			switch v := c.logger().(type) {
+			case ContextLogger:
+				v.ErrorContext(context.Background(), "error reading response body", "error", err)
```

Comment:
Why `context.Background()` here?

Reply:
Unable to get the context in the request here

Reply:
Then make `drainBody` accept `ctx`. It is private method so you can change it.

Reply:
OK~